src/client/Microsoft.Identity.Client/ManagedIdentity/CloudShellManagedIdentitySource.cs

```diff
@@ -73,14 +74,16 @@ private CloudShellManagedIdentitySource(Uri endpoint, RequestContext requestCont
             }
         }
 
-        protected override ManagedIdentityRequest CreateRequest(string resource)
+        protected override ManagedIdentityRequest CreateRequest(string resource, AcquireTokenForManagedIdentityParameters parameters)
         {
             ManagedIdentityRequest request = new ManagedIdentityRequest(HttpMethod.Post, _endpoint);
 
             request.Headers.Add("ContentType", "application/x-www-form-urlencoded");
             request.Headers.Add("Metadata", "true");
 
             request.BodyParameters.Add("resource", resource);
+            
+            ApplyClaimsAndCapabilities(request, parameters);
```

Comment:
I thought that CloudShell is out of scope?

Reply:
that is correct cloud shell is out of scope. ACI will implement token revocation after public preview of the feature.